src/NuGet.Core/NuGet.Packaging/PackagesConfigWriter.cs

```diff
@@ -18,32 +18,81 @@ namespace NuGet.Packaging
     public class PackagesConfigWriter : IDisposable
     {
         private readonly Stream _stream;
+        private readonly string _filePath;
         private bool _disposed;
-        private bool _closed;
         private NuGetVersion _minClientVersion;
         private IFrameworkNameProvider _frameworkMappings;
         private XDocument _xDocument;
 
         /// <summary>
-        /// Create a packages.config writer
+        /// Create a packages.config writer using file path
+        /// </summary>
+        /// <param name="fullPath">The full path to write the XML packages.config file into, or load existing packages.config from</param>
+        /// <param name="createNew">Whether to create a new packages.config file, or load an existing one</param>
+        public PackagesConfigWriter(string fullPath, bool createNew)
+            : this(fullPath, createNew, DefaultFrameworkNameProvider.Instance)
+        {
+        }
+
+        /// <summary>
+        /// Create a packages.config writer using file path
+        /// </summary>
+        /// <param name="fullPath">The full path to write the XML packages.config file into, or load existing packages.config from</param>
+        /// <param name="createNew">Whether to create a new packages.config file, or load an existing one</param>
+        /// <param name="frameworkMappings">Framework mappings</param>
+        public PackagesConfigWriter(string fullPath, bool createNew, IFrameworkNameProvider frameworkMappings)
+        {
+            _frameworkMappings = frameworkMappings;
+            _filePath = fullPath;
+
+            if (createNew)
+            {
+                _xDocument = new XDocument();
```

Comment:
There is no point in separating AddPackagesNode to a standalone method if it always called with _xDocument = new XDocument();

Instead you should have a create default document method.